image_tools/src/showimage.cpp

```diff
@@ -80,40 +86,34 @@ void show_image(
   }
 }
 
-int main(int argc, char * argv[])
+/// Default constructor for no command line args
+ShowImage::ShowImage(const rclcpp::NodeOptions & options)
+: Node("showimage", options)
 {
-  // Pass command line arguments to rclcpp.
-  rclcpp::init(argc, argv);
-
-  // Initialize default demo parameters
-  size_t depth = rmw_qos_profile_default.depth;
-  rmw_qos_reliability_policy_t reliability_policy = rmw_qos_profile_default.reliability;
-  rmw_qos_history_policy_t history_policy = rmw_qos_profile_default.history;
-  bool show_camera = true;
-  std::string topic("image");
-
-  // Force flush of the stdout buffer.
-  // This ensures a correct sync of all prints
-  // even when executed simultaneously within a launch file.
-  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
-
-  // Configure demo parameters with command line options.
-  if (!parse_command_options(
-      argc, argv, &depth, &reliability_policy, &history_policy, &show_camera, nullptr, nullptr,
-      nullptr, nullptr, &topic))
-  {
-    return 0;
+  std::vector<std::__cxx11::basic_string<char>> args = options.arguments();
+  int argc = options.arguments().size();
+  const char **argv = new const char*[args.size()];
```

Comment:
Converting to `char**` seems unnecessary. Since `setup()` is converting back to a `std::vector<std::string>` anyways, I would just change the signature of the function to accept a vector of strings.